packages/react/src/authentication/hooks/useSocialLogin.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  type BlackoutError,
+  type Config,
+  postAccountLink,
+  type PostAccountLinkData,
+  postSocialLogin,
+  type PostSocialLoginData,
+} from '@farfetch/blackout-client';
+import { useCallback, useReducer, useRef } from 'react';
+
+const actionTypes = {
+  PostUserSocialLoginRequest: 'POST_USER_SOCIAL_LOGIN_REQUEST',
+  PostUserSocialLoginSuccess: 'POST_USER_SOCIAL_LOGIN_SUCCESS',
+  PostUserSocialLoginFailure: 'POST_USER_SOCIAL_LOGIN_FAILURE',
+  PostAccountLinkRequest: 'POST_ACCOUNT_LINK_REQUEST',
+  PostAccountLinkSuccess: 'POST_ACCOUNT_LINK_SUCCESS',
+  PostAccountLinkFailure: 'POST_ACCOUNT_LINK_FAILURE',
+} as const;
+
+type State = {
+  isLoading: boolean;
+  error: BlackoutError | null;
+  result?: object;
+  currentRequestId: number;
+} | null;
+
+type FetchActionBase = {
+  meta: {
+    requestId: number;
+  };
+};
+
+type PostUserSocialLoginRequestAction = FetchActionBase & {
+  type: typeof actionTypes.PostUserSocialLoginRequest;
+};
+
+type PostUserSocialLoginSuccessAction = FetchActionBase & {
+  type: typeof actionTypes.PostUserSocialLoginSuccess;
+  payload: object;
+};
+
+type PostUserSocialLoginFailureAction = FetchActionBase & {
+  type: typeof actionTypes.PostUserSocialLoginFailure;
+  payload: BlackoutError;
+};
+
+type PostAccountLinkRequestAction = FetchActionBase & {
+  type: typeof actionTypes.PostAccountLinkRequest;
+};
+
+type PostAccountLinkSuccessAction = FetchActionBase & {
+  type: typeof actionTypes.PostAccountLinkSuccess;
+  payload: object;
+};
+
+type PostAccountLinkFailureAction = FetchActionBase & {
+  type: typeof actionTypes.PostAccountLinkFailure;
+  payload: BlackoutError;
+};
+
+type Action =
+  | PostUserSocialLoginFailureAction
+  | PostUserSocialLoginRequestAction
+  | PostUserSocialLoginSuccessAction
+  | PostAccountLinkFailureAction
+  | PostAccountLinkRequestAction
+  | PostAccountLinkSuccessAction;
+
+const initialState: State = null;
+
+function reducer(state: State, action: Action) {
```

Comment:
Please move this logic to the redux package as well (applies to all redux logic).